src/methods.rs

```diff
@@ -440,6 +455,18 @@ fn lint_or_fun_call(cx: &LateContext, expr: &Expr, name: &str, args: &[P<Expr>])
     }
 }
 
+/// Checks for the `CLONE_ON_COPY` lint.
+fn lint_clone_on_copy(cx: &LateContext, expr: &Expr, name: &str, args: &[P<Expr>]) {
+    if args.len() == 1 && name == "clone" {
+        let ty = cx.tcx.expr_ty(expr);
+        let parameter_environment = cx.tcx.empty_parameter_environment();
+
+        if !ty.moves_by_default(&parameter_environment, expr.span) {
```

Comment:
@oli-obk now it does :smile:
@Manishearth I’m not sure what you mean. This has nothing to do with deriving traits.